exercises/Toluwalase_Agoro/ex01_basics/src/main.cpp

```diff
@@ -35,99 +68,83 @@ int main() {
   and gets input from user so a to open
   input file
   */
-  std::vector <std::string> vectorOfWords{};
-  std::vector <int> vectorOfCounts{};
-  std::vector <int> vectorOfOrderToRead{};
-  char filePathToInput[100];
+  
+
+  char filePathToInput[150];
   std::cout << "Please input a file path to input file :";
-  std::cin.getline(filePathToInput, 100);
+  std::cin.getline(filePathToInput, 150);
   std::ifstream fileToRead;
   fileToRead.open(filePathToInput);
+  std::vector <std::pair<int, std::string>> vectorOfCountsAndWords;
 
   if (!fileToRead) {
     /* This if block checks if file was opened correctly
-    and if it was not program is terminated
+    and if not program is terminated
     */
     std::cout << "File could not be opened" << std::endl;
     std::cout << "Ensure that file path is valid" << std::endl;
   }
+  std::string holderString;
+  auto myPredictor = [&holderString](std::pair<int, std::string>& pairOfCountAndWord)
+  {
+    return (pairOfCountAndWord.second == holderString);
+  };
+
   while (!fileToRead.eof()) {
     /* This while block reads input file and adds formatted words to
     VectorOfWords until the end of file has been reached. This block
     has an if , else if block that determines whether word is more than
     4 letters and already exists in VectorOfWords. if it does it increments
     the count in VectorOfCounts otherwise it is added to VectorOfWords
     */
-    std::string holderString{ };
+    
     holderString = getWord(fileToRead);
-    auto location = std::find(vectorOfWords.begin(), vectorOfWords.end(), holderString);
-    if (holderString.size() > 4 && location == vectorOfWords.end()) {
-      vectorOfWords.push_back(holderString);
-      vectorOfCounts.push_back(1);
+    auto location = std::find_if(vectorOfCountsAndWords.begin(),vectorOfCountsAndWords.end(), myPredictor);
+    if (holderString.size() > 4 && location == vectorOfCountsAndWords.end()) {
+      vectorOfCountsAndWords.push_back(std::make_pair(1,holderString));
     }
     else {
-      if (holderString.size() > 4 && location != vectorOfWords.end())
+      if (holderString.size() > 4 && location != vectorOfCountsAndWords.end())
       {
-        vectorOfCounts[std::distance(vectorOfWords.begin(), location)] += 1;
+        vectorOfCountsAndWords[std::distance(vectorOfCountsAndWords.begin(), location)].first += 1;
       }
     }
   }
   fileToRead.close();
-  int maxCount{ 0 };
-  /*Next Block of code implements a variation of counting
-  sort  on vectorOfCounts to produce filled vectorOfOrderToRead*/
-  for (auto& count : vectorOfCounts) {
-    if (count > maxCount) {
-      maxCount = count;
-    }
-  }
-  std::vector <std::vector <int>> vectorForSort;
-  for (int i = 0; i < maxCount; i++) {
-    std::vector <int> emptyVector;
-    vectorForSort.push_back(emptyVector);
-  }
+  //countingSort(vectorOfCountsAndWords);
```

Comment:
Please can you add a comment just above this line to say that alternative counting sort method has been implemented that is faster than most implementations of the std::sort algorithm. This will indicate why you've left it commented out
I say "most implementations" because the C++ standard library doesn't mandate how std::sort has to be implemented so in theory the Microsoft version (for example) could use counting sort - but I know in reality this isn't the case.